src/modules/Dashboard/features/Participant/Activities/Activities.tsx

```diff
@@ -128,6 +134,12 @@ export const Activities = () => {
     );
   };
 
+  const getClickHandler = () => {
+    if (!subjectId || !appletId || !canAccessData) return undefined;
+
+    return onClickItemHandler;
+  };
```

Comment:
If looks like the actual handler itself has an early return statement that makes the 2/3 of the same checks.

I wonder if it would be more conventional to just update [those lines](https://github.com/ChildMindInstitute/mindlogger-admin/pull/1761/files#diff-55e1e9f202a8f57998166fcac43a4bb07de7a8d708ad50847b1675c7c48033f8R114-R115) to be something like…

```jsx
const handleClickItem = (activityId: string) => {
  if (!subjectId || !appletId || !canAccessData) return;
  // …rest of fn…
```

… compared to adding a new wrapper function that conditionally returns a handler on the basis of mostly duplicate checks.

Reply:
I'm going to remove the check on the handler instead, I explicitly wanted to return `undefined` for those cases so that the cursor would change back to a pointer in the `ActivityCard` component (e.g. by having `onClick` be `undefined` as opposed to `() => void`)